TST: SAMPProxyError Private-key expired or invalid

@embray , didn't you fix this? People keep reporting this on unrelated PR and I see that cron jobs are affected too. If the fix didn't work, I propose we just apply `pytest.mark.skipif` on CI for it.

```
_____________ ERROR at teardown of TestWebProfile.test_web_profile _____________

cls = <class '_pytest.runner.CallInfo'>
func = <function call_runtest_hook.<locals>.<lambda> at 0x7fc3e0184ee0>
when = 'teardown'
reraise = (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)

>   ???

_pytest/runner.py:311: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
_pytest/runner.py:255: in <lambda>
    ???
pluggy/hooks.py:286: in __call__
    ???
pluggy/manager.py:93: in _hookexec
    ???
pluggy/manager.py:84: in <lambda>
    ???
_pytest/threadexception.py:90: in pytest_runtest_teardown
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   pytest.PytestUnhandledThreadExceptionWarning: Exception in thread Thread-269
E   
E   Traceback (most recent call last):
E     File "astropy/samp/utils.py", line 62, in __call__
E     File "xmlrpc/client.py", line 1109, in __call__
E     File "xmlrpc/client.py", line 1450, in __request
E     File "xmlrpc/client.py", line 1153, in request
E     File "xmlrpc/client.py", line 1169, in single_request
E     File "xmlrpc/client.py", line 1341, in parse_response
E     File "xmlrpc/client.py", line 655, in close
E   xmlrpc.client.Fault: <Fault 5: 'Private-key ... expired or invalid.'>
E   
E   During handling of the above exception, another exception occurred:
E   
E   Traceback (most recent call last):
E     File "threading.py", line 932, in _bootstrap_inner
E     File "threading.py", line 870, in run
E     File "/home/runner/work/astropy/astropy/.pyinstaller/astropy_tests/samp/tests/web_profile_test_helpers.py", line 163, in _serve_forever
E       results = self.hub.pull_callbacks(self.get_private_key(), 0)
E     File "/home/runner/work/astropy/astropy/.pyinstaller/astropy_tests/samp/tests/web_profile_test_helpers.py", line 90, in pull_callbacks
E       return self._samp_hub.pullCallbacks(private_key, timeout)
E     File "astropy/samp/utils.py", line 64, in __call__
E   astropy.samp.errors.SAMPProxyError: <SAMPProxyError 5: 'Private-key ... expired or invalid.'>

_pytest/threadexception.py:75: PytestUnhandledThreadExceptionWarning
---------------------------- Captured stdout setup -----------------------------
INFO: Hub set to run with Web Profile support enabled. [astropy.samp.hub]
INFO: Hub started [astropy.samp.hub]
------------------------------ Captured log setup ------------------------------
INFO     astropy:hub.py:268 Hub set to run with Web Profile support enabled.
INFO     astropy:hub.py:402 Hub started
--------------------------- Captured stdout teardown ---------------------------
INFO: Hub is stopping... [astropy.samp.hub]
INFO: Hub stopped. [astropy.samp.hub]
---------------------------- Captured log teardown -----------------------------
INFO     astropy:hub.py:471 Hub is stopping...
INFO     astropy:hub.py:497 Hub stopped.
=========================== short test summary info ============================
ERROR astropy_tests/samp/tests/test_web_profile.py::TestWebProfile::test_web_profile
```

--- a/astropy/samp/tests/test_web_profile.py
+++ b/astropy/samp/tests/test_web_profile.py
@@ -10,6 +10,8 @@ import threading
 import tempfile
 from urllib.request import Request, urlopen
 
+import pytest
+
 from astropy.utils.data import get_readable_fileobj
 
 from astropy.samp import SAMPIntegratedClient, SAMPHubServer
@@ -21,11 +23,14 @@ from astropy.samp import conf
 
 from .test_standard_profile import TestStandardProfile as BaseTestStandardProfile
 
+CI = os.environ.get('CI', False) == "true"
+
 
 def setup_module(module):
     conf.use_internet = False
 
 
+@pytest.mark.skipif(CI, reason="Flaky on CI")
 class TestWebProfile(BaseTestStandardProfile):
 
     def setup_method(self, method):

